src/k8s-engine/wsl.ts

```diff
@@ -1207,8 +1230,25 @@ export default class WSLBackend extends events.EventEmitter implements K8s.Kuber
             const rotateConf = LOGROTATE_K3S_SCRIPT.replace(/\r/g, '')
               .replace('/var/log', logPath);
 
+            if (this.cfg?.experimentalHostResolver) {
+              console.log(`launching experimental DNS host-resolver`);
+              try {
+                this.resolverHostProcess.start();
+              } catch (error) {
+                console.error('Failed to run host-resolver vsock-host process:', error);
+              }
+              await this.writeFile('/etc/init.d/host-resolver', SERVICE_SCRIPT_HOST_RESOLVER, { permissions: 0o755 });
```

Comment:
Any reason to not always write the file out? (It _does_ need to before we run `rc-update` on it, though)